apps/financial-aid/web-veita/src/routes/ApplicationsOverview/applicationsOverviewProcessed.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useState } from 'react'
+import { useRouter } from 'next/router'
+import { useLazyQuery } from '@apollo/client'
+
+import { Text, Box, Pagination } from '@island.is/island-ui/core'
+import {
+  ApplicationsTable,
+  FilterPopover,
+} from '@island.is/financial-aid-web/veita/src/components'
+import {
+  ApplicationState,
+  ApplicationPagination,
+  applicationPageSize,
+} from '@island.is/financial-aid/shared/lib'
+import { ApplicationFilterQuery } from '@island.is/financial-aid-web/veita/graphql/sharedGql'
+import { navigationItems } from '@island.is/financial-aid-web/veita/src/utils/navigation'
+import { container } from './applicationsOverviewProcessed.css'
+
+interface Filters {
+  selectedStates: ApplicationState[]
+  selectedMonths: number[]
+}
+
+export const ApplicationsOverviewProcessed = () => {
+  const router = useRouter()
+
+  const [currentPage, setCurrentPage] = useState<number>(
+    router?.query?.page ? parseInt(router.query.page as string) : 1,
+  )
+  const [filters, setFilters] = useState<Filters>({
+    selectedStates: router?.query?.state
+      ? ((router?.query?.state as string).split(',') as ApplicationState[])
+      : [],
+    selectedMonths: router?.query?.month
+      ? (router?.query?.month as string).split(',').map(Number)
+      : [],
+  })
+
+  const onChecked = (item: ApplicationState | number, checked: boolean) => {
+    const filtersCopy = { ...filters }
+
+    if (typeof item === 'number') {
+      checked
+        ? filters.selectedMonths.push(item)
+        : filters.selectedMonths.splice(filters.selectedMonths.indexOf(item), 1)
+    } else {
+      checked
+        ? filters.selectedStates.push(item)
+        : filters.selectedStates.splice(filters.selectedStates.indexOf(item), 1)
+    }
+
+    setFilters(filtersCopy)
+  }
+
+  const setQuery = (page: number, clearFilters?: boolean) => {
+    const query = new URLSearchParams()
+    query.append('month', clearFilters ? '' : filters.selectedMonths.join(','))
+    query.append('state', clearFilters ? '' : filters.selectedStates.join(','))
+    query.append('page', page.toString())
+
+    router.push({ search: query.toString() })
+
+    setCurrentPage(page)
+    if (clearFilters) {
+      setFilters({ selectedMonths: [], selectedStates: [] })
+    }
+  }
+
+  const [getApplications, { data, error }] = useLazyQuery<{
+    filterApplications: ApplicationPagination
+  }>(ApplicationFilterQuery, {
+    fetchPolicy: 'no-cache',
+    errorPolicy: 'all',
+  })
+
+  const currentNavigationItem =
+    navigationItems.find((i) => i.link === router.pathname) ||
+    navigationItems[0]
+
+  useEffect(() => {
+    getApplications({
```

Comment:
I feel like we don't have one single source of truth for the state of the filers. We are using both states and getting it from the query. If the user manually changes the query in the URL it will call `getApplications` with old filter data? I'm I mistaken?

Can we get by not using state and use the router as the source of truth?

Reply:
If the user manually changes the query and presses enter then the page will reload and the states are set from the query. If the user changes the query but does not press enter then nothing will happen. So getApplications can't be triggered by the user in that way. 

But I agree about the source of truth. I feel like using the query to manage the state is a bit unconventional so I am going to change so that the React states are the source of truth and the query string is just there to be displayed and get the initial states. So we will no longer listen to the changes to the Router on useEffect